impeller/entity/contents/filters/gaussian_blur_filter_contents.cc

```diff
@@ -207,19 +205,48 @@ std::optional<Snapshot> DirectionalGaussianBlurFilterContents::RenderFilter(
     Command cmd;
     cmd.label = SPrintF("Gaussian Blur Filter (Radius=%.2f)",
                         transformed_blur_radius_length);
+    cmd.BindVertices(vtx_buffer);
+
     auto options = OptionsFromPass(pass);
     options.blend_mode = BlendMode::kSource;
-    cmd.pipeline = renderer.GetGaussianBlurPipeline(options);
-    cmd.BindVertices(vtx_buffer);
+    auto input_descriptor = input_snapshot->sampler_descriptor;
+    auto source_descriptor = source_snapshot->sampler_descriptor;
+    switch (tile_mode_) {
+      case Entity::TileMode::kDecal:
+        cmd.pipeline = renderer.GetGaussianBlurDecalPipeline(options);
+        break;
+      case Entity::TileMode::kClamp:
+        cmd.pipeline = renderer.GetGaussianBlurPipeline(options);
+        input_descriptor.width_address_mode = SamplerAddressMode::kClampToEdge;
+        input_descriptor.height_address_mode = SamplerAddressMode::kClampToEdge;
+        source_descriptor.width_address_mode = SamplerAddressMode::kClampToEdge;
+        source_descriptor.height_address_mode =
+            SamplerAddressMode::kClampToEdge;
+        break;
+      case Entity::TileMode::kMirror:
+        cmd.pipeline = renderer.GetGaussianBlurPipeline(options);
+        input_descriptor.width_address_mode = SamplerAddressMode::kMirror;
+        input_descriptor.height_address_mode = SamplerAddressMode::kMirror;
+        source_descriptor.width_address_mode = SamplerAddressMode::kMirror;
+        source_descriptor.height_address_mode = SamplerAddressMode::kMirror;
+        break;
+      case Entity::TileMode::kRepeat:
+        cmd.pipeline = renderer.GetGaussianBlurPipeline(options);
+        input_descriptor.width_address_mode = SamplerAddressMode::kRepeat;
+        input_descriptor.height_address_mode = SamplerAddressMode::kRepeat;
+        source_descriptor.width_address_mode = SamplerAddressMode::kRepeat;
+        source_descriptor.height_address_mode = SamplerAddressMode::kRepeat;
+        break;
```

Comment:
Oh yeah, we should totally be doing this! This has come up a couple of times before when the blur was under pressure.